### DISCUSSION TITLE
Using ef core in application layer.

### DISCUSSION THREAD

**Discussion by DimaOshchepkov** [jasontaylordev/CleanArchitecture#1255] [Q&A] [answered]
Hello! I'm learning Clean Architecture and I'm a bit confused. Most resources state that the Application layer should be independent of frameworks like Entity Framework Core. However, in your project, EF Core seems to be used within the Application layer. Could you please explain the reasoning behind this decision? Perhaps this represents a more pragmatic approach to Clean Architecture than what's typically described in theory?

**Comment by stho01** [comment, marked as answer]
This is explained here https://github.com/jasontaylordev/CleanArchitecture/wiki/ADR-001-Use-EFCore-In-Application-Layer
